src/cpu/core_dyn_x86.cpp

```diff
@@ -263,19 +263,28 @@ static void maybe_copy_host_fpu_to_dh()
 {
 	if (dyn_dh_fpu.state_used) {
 		gen_dh_fpu_save();
+		dyn_dh_fpu.state_copied_to_normal = false;
 	}
 }
 
-static void copy_dh_fpu_to_normal()
+static void maybe_copy_dh_fpu_to_normal()
 {
-	// The host's FPU should already be copied to the dh FPU state at this point
+	if (dyn_dh_fpu.state_copied_to_normal) {
+		LOG_MSG("FPU: skipping dh to normal because dh hasn't changed (normal FPU still valid)");
```

Comment:
All these should be debug or trace level, no? Regular users shouldn't see things like these in the logs. Also, what if these get called frequently? We don't want to spam the logs and slow things down.

Same comment for all log statements in this file.

Reply:
(drop them in the latest PR)